Server/Hubs/CircuitConnection.cs

```diff
@@ -440,5 +444,64 @@ private void ProcessMessages()
                 }
             }
         }
+
+        public Task WOL(string deviceId)
+        {
+            var device = _dataService.GetDevice(deviceId);
+            if (device == null)
+            {
+                return Task.CompletedTask;
+            }
+
+            if (!_dataService.DoesUserHaveAccessToDevice(deviceId, _dataService.GetUserByNameWithOrg(User.UserName)))
+            {
+                return Task.CompletedTask;
+            }
+
+            var devices = _dataService.GetDevicesForUser(User.UserName);
+            var ActiveDevices = devices.Where(c => c.DeviceGroup == device.DeviceGroup && c.OrganizationID == device.OrganizationID).Select(s => s.ID).ToArray();
+
+            var connections = GetActiveClientConnections(ActiveDevices);
+            foreach (var connection in connections)
+            {
+                string[] MACs = device.MACAddresses.Split(new char[] { ',' }, StringSplitOptions.RemoveEmptyEntries);
+                foreach (var mac in MACs) {
+                    _agentHubContext.Clients.Client(connection.Key).SendAsync("WOL", mac);
+                }
+
+                break;  // Delete the line so that all active devices will send the Magic packet. 
```

Comment:
We shouldn't have branching logic that requires source code changes and recompilation.